apps/finance/app/src/components/Transfers.js

```diff
@@ -29,236 +30,220 @@ const TRANSFER_TYPES = [
-              true,
-              { rounding: 5 }
-            )
-            return `${formatDate(
-              date
-            )},${entity},${reference},${`${formattedAmount} ${symbol}`}`
-          }
-        )
-      )
-      .join('\n')
-    return window.encodeURI(csvContent)
-  }
-  getCsvFilename = () => {
-    const { selectedDateRange } = this.state
-    const start = format(selectedDateRange.start, 'yyyy-MM-dd')
-    const end = format(selectedDateRange.end, 'yyyy-MM-dd')
-    return `transfers_${start}_to_${end}.csv`
-  }
-  showMoreTransfers = () => {
-    this.setState(prevState => ({
-      displayedTransfers: prevState.displayedTransfers + TRANSFERS_PER_PAGE,
-    }))
-  }
+  })
 
-  // Filter transfer based on the selected filters
-  getFilteredTransfers({
+  const handleToggleFiltersClick = React.useCallback(() => {
+    setFiltersOpened(!filtersOpened)
+  })
+  const handleTokenChange = React.useCallback(index => {
+    setSelectedToken(index)
+    setDisplayedTransfers(TRANSFERS_PER_PAGE)
+  })
+  const handleTransferTypeChange = React.useCallback(index => {
+    setSelectedTransferType(index)
+    setDisplayedTransfers(TRANSFERS_PER_PAGE)
+  })
+  const handleResetFilters = React.useCallback(() => {
+    setDisplayedTransfers(TRANSFERS_PER_PAGE)
+    setSelectedToken(0)
+    setSelectedTransferType(0)
+  })
+  const getCsvFilename = React.useCallback(() => {
+    let filename = 'transfers.csv'
+    const { start, end } = selectedDateRange
+    if (start && end) {
+      const formattedStart = format(selectedDateRange.start, 'yyyy-MM-dd')
+      const formattedEnd = format(selectedDateRange.end, 'yyyy-MM-dd')
+      filename = `transfers_${formattedStart}_to_${formattedEnd}.csv`
+    }
+    return filename
+  })
+  const showMoreTransfers = React.useCallback(() => {
+    setDisplayedTransfers(displayedTransfers + TRANSFERS_PER_PAGE)
+  })
+  const filteredTransfers = getFilteredTransfers({
     tokens,
     transactions,
     selectedToken,
     selectedTransferType,
     selectedDateRange,
-  }) {
-    const transferType = TRANSFER_TYPES[selectedTransferType]
-    return transactions.filter(
-      ({ token, isIncoming, date }) =>
-        (!selectedDateRange.start || !selectedDateRange.end ||
-          isWithinInterval(new Date(date), {
-            start: startOfDay(selectedDateRange.start),
-            end: endOfDay(selectedDateRange.end),
-          })) &&
-        (selectedToken === 0 ||
-          addressesEqual(token, tokens[selectedToken - 1].address)) &&
-        (transferType === TransferTypes.All ||
-          (transferType === TransferTypes.Incoming && isIncoming) ||
-          (transferType === TransferTypes.Outgoing && !isIncoming))
+  })
+  const symbols = tokens.map(({ symbol }) => symbol)
+  const tokenDetails = tokens.reduce(reduceTokenDetails, {})
+  const filtersActive = selectedToken !== 0 || selectedTransferType !== 0
+  const { resolve } = React.useContext(IdentityContext)
```

Comment:
What do you think about renaming `resolve` to `resolveAddress` or etc? Looks really similar to a `Promise.resolve()`, and we even use it in the context of a Promise later.